client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,166 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { Icon, arrowLeft } from '@wordpress/icons';
+import DomainPicker from '@automattic/domain-picker';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import { DomainSuggestions } from '@automattic/data-stores';
+
+/**
+ * Internal dependencies
+ */
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
+import {
+	MARKETPLACE_FLOW_ID,
+	ANALYTICS_UI_LOCATON_MARKETPLACE_DOMAIN_SELECTION,
+} from 'calypso/my-sites/plugins/marketplace/constants';
+import { Button } from '@wordpress/components';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { getProductsList, isProductsListFetching } from 'calypso/state/products-list/selectors';
+import { fillInSingleCartItemAttributes } from 'calypso/lib/cart-values';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+function Header() {
+	return (
+		<div className="marketplace-domain-upsell__header domains__header">
+			<h1 className="marketplace-domain-upsell__header marketplace-title">Choose a domain</h1>
+			<h2 className="marketplace-domain-upsell__header marketplace-subtitle">
+				Yoast SEO requires a top level domain to index your site and help you rank higher.
+			</h2>
+		</div>
+	);
+}
+
+const redirectToUseDomainFlow = (): void => {
+	const useYourDomainUrl = `/start/new-launch/domains-launch/use-your-domain?siteSlug=${ siteSubdomain?.domain }&source=${ launchFlowUrl }`;
+	redirectTo( useYourDomainUrl );
+};
+
+function DomainPickerContainer( { onDomainSelect, selectedDomain } ) {
+	return (
+		<DomainPicker
+			header={ <Header /> }
+			analyticsUiAlgo={ ANALYTICS_UI_LOCATON_MARKETPLACE_DOMAIN_SELECTION }
+			analyticsFlowId={ MARKETPLACE_FLOW_ID }
+			onDomainSelect={ onDomainSelect }
+			currentDomain={ selectedDomain }
+			hideRecommendationLabel
```

Comment:
In the context of https://github.com/Automattic/wp-calypso/pull/52081/files#r622168030
```suggestion
			showRecommendationLabel={ false }
```